packages/hop-node/src/watchers/classes/ContractBase.ts

```diff
@@ -135,7 +135,15 @@ export default class ContractBase extends EventEmitter {
   }
 
   get waitConfirmations () {
-    return globalConfig.networks[this.chainSlug]?.waitConfirmations ?? 0
+    const chainConfig = globalConfig.networks[this.chainSlug]
+    if (!chainConfig) {
+      throw new Error(`config for chain ${this.chainSlug} not found`)
+    }
+    const { waitConfirmations } = chainConfig
+    if (typeof waitConfirmations !== 'number') {
```

Comment:
Is this validation needed? IIUC, `waitConfirmations` are validated (it is a number and is `> 0`) during the watcher creation, which (I think) always occurs before this.

Reply:
same reason as other similar comment

Reply:
Makes sense & SGTM.